### DISCUSSION TITLE
error query doesn't fire again when keys change

### DISCUSSION THREAD

**Discussion by jdupont22** [TanStack/query#2676] [Q&A] [answered]
Hi,

I've a strange situation, i don't know if it's normal!
I try to load some "offers" with `useQuery(["offers, deepJson])`. 

**I have a "staleTime=20s" and the default "cacheTime=5m".**

`useQuery(["offers",{"order":"desc","page":1,"q":null,"size":10,"sort":"name","states":["ACTIVE"]}])`

I've the following result, react-query retry 3 times and failed! It's ok.

![Capture d’écran 2021-09-16 à 21 26 05](https://user-images.githubusercontent.com/20832813/133673418-4111bb9f-e280-4d7b-a01a-11a11ecc1927.png)

Now, I click on the tab "Inactive", that change the deepJson states to `"states":["INACTIVE"]`
I see in the react-query devtool the params change, but the request is not trigger!!!

**There is a "disabled" query with this keys, but i would imagine that a query in error will retry the next time? it's based on the cache expiration?**

![Capture d’écran 2021-09-16 à 21 26 12](https://user-images.githubusercontent.com/20832813/133673432-febeaca3-5708-4ee0-92a2-1f2ecfafeaa8.png)

Can you explain to me this current behavior? When the query return 200 OK, switching tab work perfectly!

Thx

**Comment by TkDodo** [comment]
it would be easier if you could come up with a minimal codesandbox reproduction of your situation please

**Reply by jdupont22** [reply]
Please see the [codesandbox](https://codesandbox.io/s/elated-bash-tm8gc?file=/src/App.js)

When the app launch with "fetchFail" request, the "active" status is retry 3 times, it's ok ! Please click on toggle, to relauch with the status "inactive", failed with 3 retries, it's ok ! Now, click again on toggle, none request is triggered ! 

I would expect that useQuery relaunch the failed request on any condition (staleTime, cacheTime, ...) when the keys change, no?

**Comment by TkDodo** [comment, marked as answer]
thanks for reporting. This seems to have worked correctly up until v3.19.4. I'll create an issue out of this.

Here is a slight variation of your codesandbox without window focus refetching, without retries, with a 1 second fetch delay and with devtools, so that we can better see when fetching transitions are happening: https://codesandbox.io/s/interesting-elbakyan-zz28f?file=/src/App.js

**Reply by TkDodo** [reply]
https://github.com/tannerlinsley/react-query/issues/2682

**Reply by jdupont22** [reply]
Thx for the release, work great!!!
